src/function.py

```diff
@@ -80,47 +62,30 @@ def try_delete_function_file(client: CogniteClient, external_id: str):
         logger.info(f"Did delete file {external_id}.")
 
 
-def create_and_wait(
-    client: CogniteClient,
-    name: str,
-    external_id: str,
-    function_path: str,
-    file_id: int,
-    api_key: str,
-):
+def create_and_wait(client: CogniteClient, external_id: str, config: FunctionConfig):
     logger.info(f"Will create function {external_id}")
-    function: Function = client.functions.create(
-        name=name,
+    client.functions.create(
+        name=external_id,
         external_id=external_id,
-        file_id=file_id,
-        api_key=api_key,
-        function_path=function_path,
+        folder=config.folder_path,
+        api_key=config.tenant.runtime_key,
+        function_path=config.file,
     )
-    logging.info(f"Created function {external_id}. Waiting for deployment ...")
-    wait_time_seconds = 1200  # 20 minutes
-    deployed = await_function_deployment(client, external_id, wait_time_seconds)
+    logging.info(f"Created function {external_id}. Waiting for deployment...")
+    wait_time_sec = config.deploy_wait_time_sec
+    deployed = await_function_deployment(client, external_id, wait_time_sec)
     if deployed is None:
-        print(f"Function {external_id} did not deploy within {wait_time_seconds} seconds.")
-        raise FunctionDeployTimeout(f"Function {external_id} did not deploy within {wait_time_seconds} seconds.")
+        raise FunctionDeployTimeout(f"Function {external_id} did not deploy within {wait_time_sec} seconds.")
     logging.info(f"Function {external_id} is deployed.")
     return deployed
 
 
 @retry(exceptions=(FunctionDeployTimeout, FunctionDeployError), tries=5, delay=2, jitter=2)
 def upload_and_create(client: CogniteClient, config: FunctionConfig) -> Function:
-    zip_file_name = get_file_name(config.external_id)
-    file_id = zip_and_upload_folder(client, Path(config.folder_path), zip_file_name)
     try:
-        return create_and_wait(
-            client,
-            config.external_id,
-            external_id=config.external_id,
-            function_path=config.file,
-            file_id=file_id,
-            api_key=config.tenant.runtime_key,
-        )
+        return create_and_wait(client=client, external_id=config.external_id, config=config)
     except (FunctionDeployError, FunctionDeployTimeout):
-        try_delete_function_file(client, zip_file_name)
+        try_delete_function_file(client, get_file_name(config.external_id))
```

Comment:
That's back to explicit/implicit approach. I like the change of using SDK for ZIPing, but then we have magic knowledge of how SDK generates file names... Should not SDK remove files as well then?

Reply:
Not sure why we need to delete files at all really? We could also PR experimental-client to split out the naming function, which we in turn could import here.